src/concurrency.md

```diff
@@ -33,6 +34,40 @@ fn main() {
 }
 ```
 
+[ex-rayon-any-all]: #ex-rayon-any-all
+<a name="ex-rayon-any-all"></a>
+## Check the elements of an array against a predicate in parallel
+
+[![rayon-badge]][rayon] [![cat-concurrency-badge]][cat-concurrency]
+
+The `rayon` crate provides a [`ParallelIterator`] trait which includes parallelized implementations of many of the methods from the standard library's [`Iterator`] trait.
+
+This example demonstrates using the [`rayon::any`] and [`rayon::all`] methods. [`rayon::any`] checks in parallel whether any element of the iterator matches the predicate, and returns as soon as one is found. [`rayon::all`] checks in parallel whether all elements of the iterator match the predicate, and returns as soon as a non-matching element is found.
+
+```rust
+extern crate rayon;
+
+use rayon::prelude::*;
+
+fn main() {
+    let mut vec = vec![2, 4, 6, 8];
+
+    let any_odd = vec.par_iter().any(|n| (*n % 2) != 0);
```

Comment:
I'd suggest to write these just as a single line assertions
```rust
assert!(!vec.par_iter().any(|n| (*n % 2) != 0));
```
 this would be slightly more concise

Reply:
Personally I think it's slightly more readable to break up the line and give this a name, but I'm happy to change it.